File: frontend/src/types/sessions.ts
/** 세션 도메인 타입: Session, AudioFile 인터페이스. */
export type SessionStatus = "pending" | "processing" | "completed";

export interface Session {
  id: string;
  name: string;
  deviceType: string;
  status: SessionStatus;
  fileCount: number;
  progress: number;
  score: number | null;
  createdAt: string;
  userId?: string | null;
}

export type FileStatus = "wip" | "pending" | "done";

export interface AudioFile {
  id: string;
  sessionId: string;
  filename: string;
  duration: string;
  sampleRate: string;
  status: FileStatus;
  audioUrl?: string;
}

File: backend/app/models/sessions.py
"""세션 도메인 모델: SessionResponse, AudioFileResponse."""
from typing import Optional
from app.models.common import CamelModel


class SessionResponse(CamelModel):
    id: str
    name: str
    device_type: str
    status: str
    file_count: int
    progress: int
    score: Optional[float] = None
    created_at: str


class AudioFileResponse(CamelModel):
    id: str
    session_id: str
    filename: str
    duration: str
    sample_rate: str
    status: str
    audio_url: Optional[str] = None
